src/components/common/SketchBox.js

```diff
@@ -27,17 +41,17 @@ class SketchBox extends React.Component {
         </Link>
         <hr className="sketch-divider" />
         <Row className="sketch-box-body">
-          <Col className="p-2 text-center" onClick={this.props.editFunc}>
-            <FontAwesomeIcon className="fa-lg" icon={faEdit} />
-          </Col>
-          <div className="sketch-button-divider" />
-          <Col className="p-2 text-center" onClick={this.props.downloadFunc}>
-            <FontAwesomeIcon className="fa-lg" icon={faDownload} />
-          </Col>
-          <div className="sketch-button-divider" />
-          <Col className="p-2 text-center" onClick={this.props.deleteFunc}>
-            <FontAwesomeIcon className="fa-lg" icon={faTrashAlt} />
-          </Col>
+          {
+            buttonData.filter(data => data.func)
+            .map(data => (
+              <Col className="p-2 text-center" onClick={data.func}>
+                <FontAwesomeIcon className="fa-lg" icon={data.icon} />
+              </Col>
+            ))
+            .reduce((acc, curr) => {
+              return acc.length > 0 ? [...acc, (<div className="sketch-button-divider" />), curr] : [curr];
+            }, [])
+          }
```

Comment:
This is some really smart code, and it looks clean, but each child in a list should have its own `key`.

```suggestion
            .map((data, idx) => (
              <Col className="p-2 text-center" onClick={data.func} key={idx}>
                <FontAwesomeIcon className="fa-lg" icon={data.icon} />
              </Col>
            ))
            .reduce((acc, curr) => {
              return acc.length > 0 ? [...acc, (<div className="sketch-button-divider" />), curr] : [curr];
            }, [])
          }
```